addons/block_code/ui/block_canvas/block_canvas.gd

```diff
@@ -115,3 +115,20 @@ func find_snaps(node: Node) -> Array:
 			snaps.append_array(find_snaps(c))
 
 	return snaps
+
+
+func set_scope(scope: String):
+	for block in _window.get_children():
+		if block is EntryBlock:
+			var entry_block = block as EntryBlock
+
+			if scope != entry_block.get_entry_statement():
+				entry_block.modulate = Color(0.5, 0.5, 0.5, 1)
```

Comment:
As a general advice, at least in languages that have semantic indentation (Python, GDScript) it is better to return early to avoid nesting and thus very indented code. In a for loop you can do that with break/continue. In a function you can do that with return. Example:
```
func set_scope(scope: String):
	for block in _window.get_children():
		if block is not EntryBlock:
			continue
		var entry_block = block as EntryBlock
		(...)
```